pkg/cloud/cloud.go

```diff
@@ -1835,6 +1843,62 @@ func (c *cloud) getInstance(ctx context.Context, nodeID string) (*types.Instance
 	}
 }
 
+// GetInstancesPatching returns the instance info associated with each node ID in `nodeIDs` and uses pagination
+// to get instances for large clusters. The instances are also described in batches of size up to `maxInstancesDescribed`.
+func (c *cloud) GetInstancesPatching(ctx context.Context, nodeIDs []string) ([]*types.Instance, error) {
```

Comment:
Is there ever a "one bad instance spoils the bunch" type scenario here? what if an instance is one of those weird ones like EKS's self-managed feature?